File: leet_0500_keyboard_row.ts
function findWords(words: string[]): string[] {
    const res: string[] = [];
    for (const w of words) {
        if (checker("qwertyuiop", w) || checker("asdfghjkl", w) || checker("zxcvbnm", w)) {
            res.push(w);
        }
    }
    return res;
}

function checker(charset: string, word: string): boolean {
    for (const c of word) {
        if (charset.indexOf(c.toLowerCase()) === -1) {
            return false;
        }
    }
    return true;
}

File: leet_0500_keyboard_row.go
func findWords(words []string) []string {

	var res []string
	for _, w := range words {
		if checker("qwertyuiop", w) || checker("asdfghjkl", w) || checker("zxcvbnm", w) {
			res = append(res, w)
		}
	}
	return res
}

func checker(charset string, word string) bool {

	for _, c := range word {
		if strings.Index(charset, string(unicode.ToLower(c))) == -1 {
            fmt.Println(string(c), word, charset)
			return false
		}
	}
	return true
}
